components/molecules/ContributorProfileHeader/contributor-profile-header.tsx

```diff
@@ -4,13 +4,36 @@ import RainbowBg from "img/rainbow-cover.png";
 import Button from "components/atoms/Button/button";
 import Link from "next/link";
 import { MarkGithubIcon } from "@primer/octicons-react";
+import { User } from "@supabase/supabase-js";
 
 interface ContributorProfileHeaderProps {
   avatarUrl?: string;
-  githubName?: string;
+  githubName: string;
   isConnected: boolean;
+  isFollowing: boolean;
+  handleFollow: Function;
+  handleUnfollow: Function;
+  user: User | null;
+  username: string | undefined;
 }
-const ContributorProfileHeader = ({ avatarUrl, githubName, isConnected }: ContributorProfileHeaderProps) => {
+const ContributorProfileHeader = ({
+  avatarUrl,
+  githubName,
+  isConnected,
+  isFollowing,
+  handleFollow,
+  handleUnfollow,
+  username,
+  user
+}: ContributorProfileHeaderProps) => {
+  const handleClick = () => {
+    if (isFollowing) {
+      if (handleUnfollow) handleUnfollow();
+    }
+    if (!isFollowing) {
+      if (handleFollow) handleFollow();
+    }
```

Comment:
another one:

```suggestion
    const handleAction = isFollowing ? handleUnfollow : handleFollow;
    if (handleAction) handleAction();
```

This might not the best....

Reply:
have to disagree with this. while its more concise, that comes at the cost fo readability

Reply:
have to disagree with this. while its more concise, that comes at the cost fo readability

Reply:
💯 Definitely.